tests/test_charm.py

```diff
@@ -53,7 +59,9 @@ def test_password_is_required_when_username_is_set(self):
         expected = ['prometheus-image-password']
         self.assertEqual(missing, expected)
 
-    def test_alerting_config_is_updated_by_alertmanager_relation(self):
+    @patch('charm.PrometheusCharm._get_image_tag')
+    def test_alerting_config_is_updated_by_alertmanager_relation(self, _get_image_tag_mock):
+        _get_image_tag_mock.return_value = 'v2.20.1'
```

Comment:
This is definitely a sign that we are layering things in a poor way, given that this ends up impacting every test.
Is there a way to split out how this is done, so it doesn't affect the core logic?

Reply:
Again, I agree and didn't particularly like this. However, since `pod_spec_set()` requires an image path (which requires a valid tag) and the charm ends up calling it in these other tests when most (maybe all?) hooks are triggered, I wasn't sure of how to do this without completely overhauling the charm itself.

@balbirthomas , do you have thoughts on this?